### PR TITLE
Added "Sentience" rare random event

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Due to a combination of radiation and water supply contamination,
stations have been reporting animals gaining self awareness.
/:cl:

Picks a random /mob/living/simple_animal on station that has no mind,
and gives it to a ghost who signs up, along with giving them a decent
chunk of health so they don't die instantly.

Note that this does include bots, which I've kept because it's hilarious
that the Mulebot spontaneously develops opinions.

Cockroaches now take 1 damage when stepped on, enough to kill a normal
cockroach, but not enough to kill a sentient one.

### CODE REVIEW

**Comment by Shadowlight213** [comment]
God no. Randomly making bots sentient, including securitrons, and then basically giving them antag and a bunch of health is horrible.
I would prefer not to have beepsky space me.

**Comment by Arianya** [comment]
@Shadowlight213 How does it give them antag? Reading the fluff, my reading of it would put you under normal escalation rules, similar to a purged silicon.

File: code/modules/events/sentience.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	if(!SA || !SG)
+		return FALSE
+
+	SA.key = SG.key
+	SA.languages_spoken |= HUMAN
+	SA.languages_understood |= HUMAN
+	SA.sentience_act()
+
+	SA.maxHealth = min(SA.maxHealth, 200)
```
Comment (XDTM) [inline code/modules/events/sentience.dm]: You probably meant max here, all this does is reduce health of megafauna if it gains sentience.

**Comment by AdamElTablawy** [comment]
really hoping this doesn't lead to sapient collossi

additionally, boo, this ruins sapience potions

also probably leads to a lot of balance concerns;if a spider nurse randomly gains sapience that's gg

**Comment by AdamElTablawy** [comment]
also if the mobs are nonantag couldn't this possibly lead to a blob spore randomly gaining sapience during blob and immediately trying to kill the core or otherwise help fight the blob with free movement through the blob

**Comment by AdamElTablawy** [comment]
and finally-

fuck no no sapient mulebots, do you remember pAI mules

**Comment by XDTM** [comment]
colossi would need to be on station first

this is mega rare, and sentience potions still give servitude and shit

there aren't spider nurses not made by xenobio anyway

blob spores are ultra weak anyway

**Comment by AdamElTablawy** [comment]
doesn't the spider random event produce spider nurses

**Comment by AdamElTablawy** [comment]
also-consider sapient ED-209 with no click cooldown validhunting

or worse, emagged sapient ED-209

**Comment by XDTM** [comment]
only normal ones afaik, and if both events fire, and a nurse spider of all animals gets picked, it deserves to happen

ED-209 isn't THAT strong

**Comment by AdamElTablawy** [comment]
there's a reason wiz random change lets him turn into an ed-209

they have no click cooldown, and don't get stunned, so they can just spamclick tazers like a fucking ranged holoparasite

**Comment by AdamElTablawy** [comment]
honestly this wouldn't really be an issue 99% of the time, the worst you'd get is sapient runtime maybe dragging people out of the way and triggering clockcult sigils and laughing hysterically or possibly a cockroach being annoying, but that 1% of the time is probably going to be extremely cancer

it's a rare event, sure, but still

**Comment by XDTM** [comment]
ED-209 are made very rarely, and this event is very rare and unreliable as well. There's no real room for abusing this, and it probably happens more often than that due to admin shenanigans.

**Comment by XDTM** [comment]
200 health is a bit much, though, @coiax. It should be 30-50 so you can survive a couple of hits but not become a beepsky tank.

**Comment by AdamElTablawy** [comment]
fair enough, i guess. the chances of a rare random event hitting a rare bot on a round that lasts that long to just so happen to select an ed-209 and not some random monkey is balance enough by itself

also now i really want to see a random event monkey become sapient in virology while infected and promptly release an ungodly plague

**Comment by ExcessiveUseOfCobblestone** [comment]
purged silicon having normal escalation rules is a mistake though

**Comment by KorPhaeron** [comment]
Cool

**Comment by KorPhaeron** [comment]
Even if it did make a muderous mulebot we already have events that create swarmers/morphs/xenomorphs/blobs/space ninjas

**Comment by lzimann** [comment]
sentient mobs are unfun

**Comment by ChangelingRain** [comment]
limit this by sentience types so things that aren't valid for the potion aren't considered

**Review by ChangelingRain — DISMISSED** [review]
limit this by sentience types so that things that wouldn't be valid for the sentience potion don't get affected

**Comment by KorPhaeron** [comment]
That would limit it to mice and slimes. Boring.

**Comment by ShadowDeath6** [comment]
Mules need to be hacked into grief mode before they can grief remember

**Comment by Jazaen** [comment]
Hey, does this count in slimes? Because that would make doing xenobio problematic, since good part of station mob population is there.

**Comment by coiax** [comment]
This does count slimes! I see no reason to deny people the fun attempts of signalling they are intelligent via dance moves.

**Comment by optimumtact** [comment]
Can you fix the lore to be intelligence or something so I can merge this without Cheridan removing my head.

**Comment by Shadowlight213** [comment]
Giving anything sentienced at least 200 health is still insane.

**Comment by ShadowDeath6** [comment]
Branch name is A+ as well this is a good PR other than beepsky having the same health as a human

**Pull request merged**